### DISCUSSION TITLE
Is there any way to use doh3 in libcurl .

### DISCUSSION THREAD

**Discussion by ag2s20150909** [curl/curl#9437] [Q&A] [answered]
Is there any way to use doh3 in libcurl .
Both Google DNS and Cloudflare DNS support doh3

https://security.googleblog.com/2022/07/dns-over-http3-in-android.html

**Comment by bagder** [comment, marked as answer]
Unclear.

It should not be that hard to get it going at least:

- HTTP/3 support is still labelled as `EXPERIMENTAL` in curl, but if you enable it in your build it runs pretty good (with some caveats)
- The DoH support in curl has not been tested for HTTP/3, but I don't see any obvious reasons why it shouldn't work, or be able to work fairly easily

I would certainly appreciate if someone investigated this a bit deeper and told us how it behaves!
